src/schema/egrid.ts

```diff
@@ -106,3 +106,13 @@ export const Location = z.enum([
 ]);
 
 export type Location = z.infer<typeof Location>;
+
+/*
+ * EgridRecord Schema
+ */
+export const EgridRecord = z.object({
+  year: z.number().int().min(2000).max(2100), //years: 2000 - 2100
+  location: Location,
+});
```

Comment:
This is a great starting point for the record with validation! Could you also add all the data fields form the `EgridModel` schema? I know this is pretty tedious so I'd recommend using AI to help you do the conversions. Also, I think it would be helpful to split the key and other data fields into the following structure:

- `EgridRecordKey`: contains the `year` and `location` fields
- `EgridRecordData`: contains all the other data fields
- `EgridRecord`: union of `EgridRecordKey` and `EgridRecordData`

This will make it easier later on to separate our key-building and data-building logic